### DISCUSSION TITLE
Error Reading Options file: What is the default log file path?

### DISCUSSION THREAD

**Discussion by benjaminplong** [ERGO-Code/HiGHS#1628] [Q&A] [answered]
I see that Highs_setLogFile() is deprecated. I am getting an error trying to read an options file, but I am not sure where to look for more information.

**Comment by jajhall** [comment]
How are you trying to pass the options? Via the command line, or from an application call?

**Reply by benjaminplong** [reply]
I am making a call to Highs_readOptions() pass a file path from source code

**Reply by jajhall** [reply]
So, you're calling from C?

What logging to you get whilst `Highs_readOptions()` is executing?

**Reply by benjaminplong** [reply]
I am calling the C++ code from Java. The HighsInt being returned is -1. I would like to know why the options file isn't loading properly, but I can't find that information.

I did a small test with my options file and the python interface errors reading the file. The file was written using Highs_writeOptions
[options.txt](https://github.com/ERGO-Code/HiGHS/files/14348956/options.txt)

**Reply by jajhall** [reply]
OK I'll try the Python interface.

I've never heard of HiGHS having a Java interface!

**Reply by benjaminplong** [reply]
A follow up point is Highs_writeOptionsDeviations works properly to write a file and then reimport it

**Reply by benjaminplong** [reply]
I have done some trial and error with the file. It seems some options written aren't supported anymore / their values can't be read. Example:

Removing the time_limit value allows small_matrix_value to be set to 3.3 instead of its default. So it looks like the parsing of time_limit is incorrect. 

 Time limit (seconds)
 [type: double, advanced: false, range: [0, inf], default: inf]
time_limit = inf

 Lower limit on |matrix entries|: values less than or equal to this will be treated as zero
 [type: double, advanced: false, range: [1e-12, inf], default: 1e-09]
small_matrix_value = 3.3

There is also an option that seems to not exist anymore:
 MIP minimum logging interval
 [type: double, advanced: false, range: [0, inf], default: 5]
mip_min_logging_interval = 5

**Reply by jajhall** [reply, marked as answer]
The short answer is "Only define the non-default values" that you want to use

My guess is that "inf" isn't being interpreted corrctly

mip_min_logging_interval is still defined

**Reply by jajhall** [reply]
Yes, `time_limit=inf` being set causes a failure

This will be fixed in `latest` when #1642 is merged
